internal/lib/lib.go

```diff
@@ -386,8 +401,8 @@ func UseServicesAPI() bool {
 	return false
 }
 
-func CheckControllerVersionCompatibility(version, maxVersion string) bool {
-	if c, err := semver.NewConstraint(fmt.Sprintf("> %s", maxVersion)); err == nil {
+func CheckControllerVersionCompatibility(version, maxVersion, cmpSign string) bool {
```

Comment:
we can change this function to take arguments like
`CheckControllerVersionCompatibility(v1, compare, v2 string)` 
removing the `"max"Version`, and putting the  compare Sign in the middle so that the implementations are eassier to read like
`CheckControllerVersionCompatibility(currentVersion, ">=",  GRBACControllerVersion)`